assets/magnify.js

```diff
@@ -0,0 +1,43 @@
+// create a container and set the full-size image as its background
+function createOverlay(image) {
+  overlay = document.createElement('div');
+  overlay.setAttribute('class', 'image-magnify__full-size');
+  overlay.setAttribute('aria-hidden', 'true');
+  overlay.style.backgroundImage = `url('${image.src}')`;
+  image.parentElement.insertBefore(overlay, image);
+  return overlay;
+};
+
+function moveWithHover(image, event, zoomRatio) {
+  // calculate mouse position
+  const ratio = image.height / image.width;
+  const container = event.target.getBoundingClientRect();
+  const xPosition = event.clientX - container.left;
+  const yPosition = event.clientY - container.top;
+  const xPercent = `${xPosition / (overlay.clientWidth / 100)}%`;
+  const yPercent = `${yPosition / ((overlay.clientWidth * ratio) / 100)}%`;
+
+  // determine what to show in the frame
+  overlay.style.backgroundPosition = `${xPercent} ${yPercent}`;
+  overlay.style.backgroundSize = `${image.width * zoomRatio}px`;
+};
+
+function magnify(image, zoomRatio) {
+  // add full-size image on top of original
+  const overlay = createOverlay(image);
+  overlay.onclick = () => overlay.remove();
+  overlay.onmousemove = (event) => moveWithHover(image, event, zoomRatio);
+  overlay.onmouseleave = () => overlay.remove();
+}
+
+function enableZoomOnHover(zoomRatio) {
+  const images = document.querySelectorAll('.image-magnify__hover');
+  images.forEach(image => {
+    image.onclick = (event) => {
+      magnify(image, zoomRatio);
+      moveWithHover(image, event, zoomRatio);
+    };
+  });
+}
+
+enableZoomOnHover(2);
```

Comment:
That could be an interesting value to tie to a setting in the editor. Can probably be done in the future if we see a lot of requests to change the zoom value.